kratos/sources/model_part_io.cpp

```diff
@@ -179,18 +179,41 @@ namespace Kratos
         std::string aux_string;
         const std::string string_to_remove = "This properties contains 0 tables";
 
-        for(PropertiesContainerType::const_iterator i_properties = rThisProperties.begin() ; i_properties != rThisProperties.end() ; i_properties++)
-        {
+        for (auto i_properties = rThisProperties.begin() ; i_properties != rThisProperties.end() ; ++i_properties) {
             std::ostringstream aux_ostream;
             (*mpStream) << "Begin Properties " << i_properties->Id() << std::endl;
             i_properties->PrintData(aux_ostream);
 
             aux_string = aux_ostream.str();
-            std::string::size_type i = aux_string.find(string_to_remove);
 
-            if (i != std::string::npos)
-            {
-                aux_string.erase(i, string_to_remove.length());
+            // We remove the line of Constitutive Laws and we add it manually. We do this because when calling the Info() method to the data_value_container it returns the address of the pointer, and we are interested in the name
+            if (i_properties->Has(CONSTITUTIVE_LAW)) {
+		// First we remove the entire line (we will add it by ourselves later) 
+                std::string::size_type it_constitutive_law_begin = aux_string.find("CONSTITUTIVE_LAW");
+                std::string::size_type it_constitutive_law_end = aux_string.find("\n", it_constitutive_law_begin);
```

Comment:
`it_constitutive_law_end` could be moved inside the if statement below.

Reply:
OK